### ISSUE TITLE
[Question] Rescheduling with load balancer on seems slower than some weeks ago

### ISSUE THREAD

**Issue by galantra** [open-spaced-repetition/fsrs4anki-helper#131]
I've noticed that rescheduling with load balancing active takes much more time than I remember it taking a couple of weeks ago. For example, in one deck, I measured ~20.5 s to reschedule just 11 cards.

(With load balancer off, rescheduling is nearly instantaneous.)

Can you confirm the slower speed? 

Do you know what the cause is? Perhaps it's related to the _disperse siblings_ function.

**Comment by L-M-Sherlock** [comment]
> I've noticed that rescheduling with load balancing active takes much more time than I remember it taking a couple of weeks ago. For example, in one deck, I measured ~20.5 s to reschedule just 11 cards.

I don't know which PR affected it. Could you share the deck with me?

**Comment by L-M-Sherlock** [comment]
> I've noticed that rescheduling with load balancing active takes much more time than I remember it taking a couple of weeks ago. For example, in one deck, I measured ~20.5 s to reschedule just 11 cards.

20.5s is too long. I haven't encounter this case in my collection. Maybe there are some marginal case that I haven't considered?

**Comment by galantra** [comment]
I am going to prepare the deck for sharing so that you will be able to investigate it if you want.

But I'm already letting you know that it seems to be an anomaly, an outlier. Because now I've tested it (specifically, rescheduling the cards reviewed in the last 14 days) on other decks, too; and on all those other decks it was as quick as I am used to. What's more, since my post I've reviewed 15 more cards from the problematic deck; now the 14-day-reschedule is as instantaneous as it is with the other decks.

So I'm going to upload the backup of the deck from the specific point in time where rescheduling is slow.

**Comment by galantra** [comment]
[The collection in question](https://github.com/open-spaced-repetition/fsrs4anki-helper/files/11757921/slow_collection.zip)

Let me emphasize that this probably doesn't deserve attention, right now, since it went away after some more reviews. I will make a comment here if it occurs again.

**Comment by L-M-Sherlock** [comment]
> [The collection in question](https://github.com/open-spaced-repetition/fsrs4anki-helper/files/11757921/slow_collection.zip)
> 
> Let me emphasize that this probably doesn't deserve attention, right now, since it went away after some more reviews. I will make a comment here if it occurs again.

`[1.449, 1.4242, 2.3924, 0.5435, 1.7644, 0.0014, 6.6995, 0.4682, 1.6665, 2.3379, 0.0958, 0.5394, 1.3677, 0.835, 0.5439, 0.8149, 2.9282, 4.6023, 0.422]` is your global parameters, which is wrong.

**Comment by galantra** [comment]
Okay, thanks for making me aware of that. They are probably from when I was trying out one of the alpha versions of the Optimizer (as you see, I hadn't reviewed the deck much in a long time).

I've generated the parameters anew ([1.8211, 1.7634, 4.6962, -0.2402, -1.6569, 0.05, 1.86, -0.3432, 1.2528, 2.269, -0.1047, 0.5289, 1.3699]
), making sure I was using https://colab.research.google.com/github/open-spaced-repetition/fsrs4anki/blob/v3.24.6/fsrs4anki_optimizer.ipynb. I can confirm that this change made rescheduling very fast again.

PS: Also, I have to correct this statement of mine:

> What's more, since my post I've reviewed 15 more cards from the problematic deck; now the 14-day-reschedule is as instantaneous as it is with the other decks.

It was not the reviews that fixed the speed of rescheduling. I had also made another change in the meantime, which was updating the scheduler JS code with newly generated parameters (from https://colab.research.google.com/github/open-spaced-repetition/fsrs4anki/blob/v3.24.6/fsrs4anki_optimizer.ipynb).
